baseplate/lib/secrets.py

```diff
@@ -356,6 +380,88 @@ def _get_data(self) -> Tuple[Dict, float]:
         return self._data
 
 
+class DirectorySecretsStore(SecretsStore):
+    """Access to secret tokens with automatic refresh when changed.
+
+    This local vault allows access to the secrets cached on disk given a path to
+    a directory. It will automatically reload the cache when it is changed. Do not
+    cache or store the values returned by this class's methods but rather get
+    them from this class each time you need them. The secrets are served from
+    memory so there's little performance impact to doing so and you will be
+    sure to always have the current version in the face of key rotation etc.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        parser: SecretParser,
+        timeout: Optional[int] = None,
+        backoff: Optional[float] = None,
+    ):  # pylint: disable=super-init-not-called
+        self.parser = parser
+        self._filewatchers = {}
+        for root, _, files in os.walk(path):
+            for f in files:
+                p = Path(root, f)
+                file_path = str(p.relative_to(path))
+                self._filewatchers[file_path] = FileWatcher(
+                    file_path, json.load, timeout=timeout, backoff=backoff
+                )
```

Comment:
:no_bell: this can be fully written with pathlib to simplify a bit

```suggestion
        root = Path(path)
        for p in root.glob("**"):
            file_path = str(p.relative_to(root))
            self._filewatchers[file_path] = FileWatcher(
                file_path, json.load, timeout=timeout, backoff=backoff
            )
```

Reply:
oh yeah, that looks much nicer, awesome